features/step_definitions/sd-step.js

```diff
@@ -20,77 +22,81 @@ module.exports = function server() {
-                },
-                json: true
-            });
-        })
-        .then((response) => {
-            Assert.oneOf(response.statusCode, [409, 201]);
+    Given(/^an existing pipeline with (.*) image and (.*) package$/,
+        { timeout: TIMEOUT }, function step(image, pkg) {
+            return this.getJwt(this.accessKey)
+                .then((response) => {
+                    this.jwt = response.body.token;
+                    this.expectedImage = image;
 
-            if (response.statusCode === 201) {
-                this.pipelineId = response.body.id;
-            } else {
-                const str = response.body.message;
-                const id = str.split(': ')[1];
+                    return request({
+                        uri: `${this.instance}/${this.namespace}/pipelines`,
+                        method: 'POST',
+                        auth: {
+                            bearer: this.jwt
+                        },
+                        body: {
+                            checkoutUrl:
+                                `git@github.com:${this.repoOrg}/${this.repoName}.git#master`
+                        },
+                        json: true
+                    });
+                })
+                .then((response) => {
+                    Assert.oneOf(response.statusCode, [409, 201]);
 
-                this.pipelineId = id;
-            }
-        })
-    );
+                    if (response.statusCode === 201) {
+                        this.pipelineId = response.body.id;
+                    } else {
+                        const str = response.body.message;
+                        const id = str.split(': ')[1];
 
-    this.When(/^the (.*) job is started$/, { timeout: TIMEOUT }, jobName =>
-        request({
-            uri: `${this.instance}/${this.namespace}/pipelines/${this.pipelineId}/jobs`,
-            method: 'GET',
-            json: true
-        }).then((response) => {
-            Assert.equal(response.statusCode, 200);
+                        this.pipelineId = id;
+                    }
+                });
+        });
 
-            for (let i = 0; i < response.body.length; i += 1) {
-                if (response.body[i].name === jobName) {
-                    this.jobId = response.body[i].id;
-                    this.image = response.body[i].permutations[0].image;
-                    this.commands = response.body[i].permutations[0].commands;
-                    break;
-                }
-            }
-            Assert.equal(this.image, this.expectedImage);
-        })
-        .then(() =>
-            request({
-                uri: `${this.instance}/${this.namespace}/builds`,
-                method: 'POST',
-                body: {
-                    jobId: this.jobId
-                },
-                auth: {
-                    bearer: this.jwt
-                },
+    When(/^the (main|tilde|hat|specify) job is started$/,
```

Comment:
I'm not sure why this file is displaying weirdly in the diff view. This is the only significant change here, because with the `.*` expression this step was also applying to one from secrets, which cucumber doesn't know how to deal with.